src/async_traits.rs

```diff
@@ -41,6 +41,56 @@ fn retryable_error(err: &DieselError) -> bool {
     }
 }
 
+// Kills the connection when this object is dropped.
+//
+// Specifically, this sets the diesel "TransactionManager" state
+// to broken, which should prevent any subsequent access to the
+// connection from succceeding.
+//
+// This aims to help avoid leaving open transactions alive
+// if an asynchronous transaction is cancelled.
+struct ConnectionKiller<'a, Conn>
```

Comment:
should this, perhaps, have a 
```suggestion
#[must_use]
struct ConnectionKiller<'a, Conn>
```
on it, to stop our future selves from accidentally making one that can't be disarmed?

Reply:
Yeah, seems reasonable! Updated.